### PR TITLE
Project 2: Add reducers and actions.

### PR DESCRIPTION
**Pull request by muhammedb8826**
### I have worked on the following
- Added Redux Toolkit   
```sh
  npm install react-redux @reduxjs/toolkit
```
- Created a (/redux) directory that will contain all my Redux logic inside the directory (/src).
- Configured a Redux store inside (/src/redux/store.js).
- Defined a slice of state for books that:
    Included an array of books (initial state: empty array)

- Defined a slice of state for categories that:
    Includes an array of categories (initial state: empty array)
    Includes a reducer that checks the status and always returns "Under construction".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,241 @@
...
+- About the Project
+  - Built With
+  - Live Demo
+- Getting Started
+- Authors
+- Future Features
+- Contributing
+- Show your support
+- Acknowledgements
+- License
```
Comment (Shedrack-Sunday) [inline README.md]: - [ ] Kindly remove this commented code from your project. These are guides and should be removed when the code has been implemented. It keeps your code clean, and  professional 👍

File: src/reportWebVitals.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    }) => {
+      getCLS(onPerfEntry);
+      getFID(onPerfEntry);
+      getFCP(onPerfEntry);
+      getLCP(onPerfEntry);
+      getTTFB(onPerfEntry);
+    });
+  }
+};
+
```
Comment (Shedrack-Sunday) [inline src/reportWebVitals.js]: - [ ] This file is not important in your codebase and had no significant value. Kindly take it out 👍

File: src/redux/books/booksSlice.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  books: [],
+  isLoading: true,
+};
+
+const booksSlice = createSlice({
+  name: 'book',
+  initialState,
+});
+
+export default booksSlice.reducer;
```
Comment (Shedrack-Sunday) [inline src/redux/books/booksSlice.js]: - [ ] Based on project requirements, you're meant to creates 2 reducers, `addBook` and `removeBook`
- [ ] You're also meant to exports the reducers:  `addBook` and `removeBook` as actions. Kindly implement both 👍 

## Instructions:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @muhammedb8826          👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 


## Highlights
✔️ Moderate use of React liberaries
✔️ Nice code implementation
  
# Required Changes ♻️

- [ ] There are a few reducers asked in the book store which you need to implement,  to have the add and remove functionality on your app
- [ ] You have some commented code which i highlighted 

_check the comment section for a detailed review_ 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md [hunk 4]
```diff
@@ -1 +1,216 @@
-# Bookstore
\ No newline at end of file
+<a name="readme-top"></a>
+
+<div align="center">
+  <!-- You are encouraged to replace this logo with your own! Otherwise you can also remove it. 
+  <img src="./images/moon.png" alt="logo" width="140"  height="auto" /> -->
```
Comment (nahnahdev) [inline README.md]: - [ OPTIONAL ] Kindly avoid leaving commented pieces of code in your codebase. It makes your codebase unclean and is not a good practice :+1:

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @muhammedb8826 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- All requested changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**